dlog/commitment/src/commitment.rs

```diff
@@ -724,7 +725,33 @@ where
                 }
                 assert_eq!(j, omegas.unshifted.len());
             }
-            (p, omega)
+
+            let mut res = DensePolynomial::<Fr<G>>::zero();
+
+            let scaled: Vec<_> = plnm_chunks
+                .par_iter()
+                .map(|(scale, segment)| {
+                    let scale = *scale;
+                    match segment {
+                        OptShiftedPolynomial::Unshifted(segment) => {
+                            let v = segment.par_iter().map(|x| scale * *x).collect();
+                            DensePolynomial::from_coefficients_vec(v)
+                        }
+                        OptShiftedPolynomial::Shifted(segment, shift) => {
+                            let mut v: Vec<_> = segment.par_iter().map(|x| scale * *x).collect();
+                            let mut res = vec![Fr::<G>::zero(); *shift];
+                            res.append(&mut v);
+                            DensePolynomial::from_coefficients_vec(res)
+                        }
+                    }
+                })
+                .collect();
+
+            for p in scaled {
+                res += &p;
+            }
```

Comment:
could this logic of reconstructing a polynomial from its chunks be abstracted in `utils/src/polynomials.rs` or something like that? 

```rust
pub trait ChunkedPolynomial: Default {
  fn add_unshifted(&mut self, scale: F, coeffs: Vec<F>)
  fn add_shifted(&mut self, scale: F, coeffs: Vec<F>)
  fn reconstruct(self) -> DensePolynomial
}
```

Reply:
Thanks, did something like this